ClinGen/clincoded
# Connects to #543. Modal for obtaining external resources (ClinVar preferred titles)

* New ClinVar VariationID add panel
![image](https://cloud.githubusercontent.com/assets/4326866/12875267/22463e4a-cd9b-11e5-901a-e13ab3f5682d.png)
![image](https://cloud.githubusercontent.com/assets/4326866/12875281/5f3f2b22-cd9b-11e5-8996-4a12fd8bfe15.png)

* VariationID add modal
![image](https://cloud.githubusercontent.com/assets/4326866/12875270/38346024-cd9b-11e5-8b5a-a44660f2398b.png)
![image](https://cloud.githubusercontent.com/assets/4326866/12875278/546a8a0c-cd9b-11e5-9cb1-e8146b74097e.png)

Testing:

1. Create GDM, add PMID
2. Create Individual to test Add ClinVar VariationID modal:
  1. Click button to ensure modal pops up
  2. Enter ClinVar VariationID into input box; ensure that the button below input is enabled
  3. Click button below input field, ensure that the cog appears, and then ensure that the correct ClinVar preferred title appears, and that the button to save becomes enabled
  4. Change the value in the input box, and ensure that the previously fetched ClinVar preferred title disappears, and the Save button is disabled
  5. Fetch again, then save
  6. Ensure that the Other Description box is disabled, you get the ClinVar ID and Preferred Title on the main form, the ClinVar ID is linked, and the Add button becomes Edit and Clear
3. Test above modal on Family and Experimental Data curation pages as well
